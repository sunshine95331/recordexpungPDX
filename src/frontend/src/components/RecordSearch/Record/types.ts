export const shortLabels = [
  "Eligible Now",
  "Eligible Future",
  "Eligible",
  "Needs Analysis",
  "Eligible?",
  "Ineligible",
  "Unknown",
];

export type ShortLabel = (typeof shortLabels)[number];

export interface ChargeData {
  case_number: string;
  ambiguous_charge_id: string;
  statute: string;
  expungement_result: any;
  expungement_rules: string;
  name: string;
  type_name: string;
  level: string;
  date: string;
  disposition: {
    status: string;
    ruling: string;
    date: string;
  };
  probation_revoked: string;
  edit_status: string;
  shortLabel?: ShortLabel;
  isExcluded?: boolean;
}

export interface CaseData {
  balance_due: number;
  birth_year: number;
  case_detail_link: string;
  case_number: string;
  charges: ChargeData[];
  citation_number: string;
  current_status: string;
  district_attorney_number: string;
  edit_status: string;
  date: string;
  location: string;
  name: string;
  violation_type: string;
}

export interface RecordData {
  total_balance_due?: number;
  cases?: CaseData[];
  errors?: string[];
  summary?: RecordSummaryData;
  questions?: QuestionsData;
}

export interface RecordSummaryData {
  total_charges: number;
  charges_grouped_by_eligibility_and_case: { [label: string]: any[] };
  county_fines: CountyFinesData[];
  total_fines_due: number;
  total_cases: number;
}

export interface CaseFineData {
  case_number: string;
  balance: number;
}

export interface CountyFinesData {
  county_name: string;
  case_fines: CaseFineData[];
  total_fines_due: number;
}

export interface ExpungementResultData {
  type_eligibility: TypeEligibilityData;
  time_eligibility?: TimeEligibilityData;
  charge_eligibility: ChargeEligibilityData;
}

export interface TypeEligibilityData {
  status: string;
  reason: string;
}

export interface TimeEligibilityData {
  status: string;
  reason: string;
  date_will_be_eligible: string;
}

export const chargeEligibilityStatuses = [
  "Eligible Now",
  "Will Be Eligible",
  "Possibly Eligible",
  "Possibly Will Be Eligible",
  "Needs More Analysis",
  "Ineligible",
  "Unknown",
] as const;

export type ChargeEligibilityStatus =
  (typeof chargeEligibilityStatuses)[number];

export interface ChargeEligibilityData {
  status: ChargeEligibilityStatus;
  label: string;
  date_to_sort_label_by: string | null;
}

export interface QuestionsData {
  [ambiguous_charge_id: string]: QuestionSummaryData;
}

export interface QuestionSummaryData {
  case_number: string;
  ambiguous_charge_id: string;
  root: QuestionData;
}

export interface QuestionData {
  question_id: string;
  text: string;
  options: { [option: string]: AnswerData };
  selection: string;
  convicted_date_string: string;
  probation_revoked_date_string: string;
}

export interface AnswerData {
  question?: QuestionData | null;
  edit?: { [key: string]: string };
}

export const CHARGE_TYPES = [
  "Civil Offense",
  "Criminal Forfeiture",
  "Dismissed Criminal Charge",
  "DUII",
  "Diverted DUII",
  "FareViolation",
  "Felony Class A",
  "Felony Class B",
  "Felony Class C",
  "Juvenile",
  "Marijuana Eligible",
  "Marijuana Eligible (Below age 21)",
  "Marijuana Violation",
  "Marijuana Ineligible",
  "Misdemeanor Class A",
  "Misdemeanor Class B or C",
  "Parking Ticket",
  "Person Felony Class B",
  "Reduced to Violation",
  "Severe Charge",
  "Sex Crime",
  "Young Offender Sex Crime, Needs More Analysis",
  "Young Offender Sex Crime, Ineligible",
  "Subsection 6",
  "Traffic Offense",
  "Traffic Violation",
  "Unclassified",
  "Violation",
];

export const CHARGE_TYPES_CONVICTED_ONLY = [
  "Felony Class A",
  "Felony Class B",
  "Felony Class C",
  "DUII",
  "Marijuana Eligible",
  "Marijuana Eligible (Below age 21)",
  "Marijuana Ineligible",
  "Misdemeanor Class A",
  "Misdemeanor Class B or C",
  "Person Felony Class B",
  "Severe Charge",
  "Sex Crime",
  "Young Offender Sex Crime, Needs More Analysis",
  "Young Offender Sex Crime, Ineligible",
  "Subsection 6",
  "Traffic Offense",
];

export const CHARGE_TYPES_DISMISSED_ONLY = [
  "Dismissed Criminal Charge",
  "Diverted DUII",
];

export const SEVERITY_LEVELS = [
  "Felony Class A",
  "Felony Class B",
  "Felony Class C",
  "Misdemeanor Class A",
  "Misdemeanor Class B or C",
  "Violation",
];
