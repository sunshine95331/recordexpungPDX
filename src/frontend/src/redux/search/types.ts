import {
  RecordData,
  RecordSummaryData,
  QuestionsData,
} from "../../components/RecordSearch/Record/types";
import { AliasData } from "../../components/RecordSearch/SearchPanel/types";

export interface SearchResponse {
  record: RecordEndpointData;
}

export interface RecordEndpointData {
  total_balance_due: number;
  cases: any[];
  errors: string[];
  summary: RecordSummaryData;
  questions: QuestionsData;
}

export const DISPLAY_RECORD = "DISPLAY_RECORD";
export const RECORD_LOADING = "RECORD_LOADING";
export const CLEAR_RECORD = "CLEAR_RECORD";
export const SELECT_ANSWER = "SELECT_ANSWER";

export interface SearchRecordState {
  loading: string;
  aliases: AliasData[];
  record?: RecordData;
  questions?: QuestionsData;
  edits?: any;
}

interface SearchRecordAction {
  type: typeof DISPLAY_RECORD | typeof RECORD_LOADING | typeof CLEAR_RECORD;
  aliases: AliasData[];
  record: RecordData;
  questions: QuestionsData;
}

export interface QuestionsAction {
  type: typeof SELECT_ANSWER;
  ambiguous_charge_id: string;
  case_number: string;
  question_id: string;
  answer: string;
  edit: any;
  date: string;
  probation_revoked_date: string;
}

export type SearchRecordActionType = SearchRecordAction | QuestionsAction;
