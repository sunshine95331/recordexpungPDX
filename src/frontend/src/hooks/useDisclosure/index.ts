// Reference: https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/

import React, { useState } from "react";

export interface DisclosureButtonProps {
  "aria-controls": string;
  "aria-expanded": boolean;
  onClick: React.MouseEventHandler;
  onKeyDown: React.KeyboardEventHandler;
}

export interface DisclosureContentProps {
  id: string;
  hidden?: boolean;
}

export default function useDisclosure(
  id = "panel--disclosure",
  isOpenToStart = false
) {
  const [disclosureIsExpanded, setIsExpanded] = useState(isOpenToStart);

  const onClick = (event: React.MouseEvent) => {
    event.preventDefault();
    setIsExpanded(!disclosureIsExpanded);
  };

  const onKeyDown = (event: React.KeyboardEvent) => {
    if (event.key === " " || event.key === "Enter") {
      event.preventDefault();
      setIsExpanded(!disclosureIsExpanded);
    }
  };

  const disclosureButtonProps: DisclosureButtonProps = {
    "aria-controls": id,
    "aria-expanded": disclosureIsExpanded,
    onClick,
    onKeyDown,
  };

  const disclosureContentProps: DisclosureContentProps = {
    id,
    hidden: !disclosureIsExpanded,
  };

  return {
    disclosureIsExpanded,
    disclosureButtonProps,
    disclosureContentProps,
  };
}
