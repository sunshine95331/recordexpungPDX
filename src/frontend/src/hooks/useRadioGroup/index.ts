import React, { useState } from "react";

interface RadioButtonProps {
  id: string;
  name: string;
  type: string;
  value: string;
  "aria-checked": boolean;
  checked: boolean;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

interface RadioLabelProps {
  htmlFor: string;
}

export interface RadioGroupWrapperProps {
  dir: string;
  role: "radiogroup";
  "aria-label": string;
}

export type MakeRadioButtonProps = (label: string) => {
  radioButtonProps: RadioButtonProps;
  radioLabelProps: RadioLabelProps;
};

export interface UseRadioGroupReturn {
  selectedRadioValue: string;
  radioGroupWrapperProps: RadioGroupWrapperProps;
  makeRadioButtonProps: MakeRadioButtonProps;
}

interface Props {
  label: string;
  initialValue?: string;
}

export default function useRadioGroup({ label, initialValue = "" }: Props) {
  const [selectedRadioValue, setSelectedRadioValue] = useState(initialValue);
  const groupLabel = label.toLowerCase();
  const prefix = groupLabel.replace(/\s/g, "-") + "-";
  const radioGroupWrapperProps: RadioGroupWrapperProps = {
    dir: "ltr",
    role: "radiogroup",
    "aria-label": label,
  };

  const makeRadioButtonProps: MakeRadioButtonProps = (label: string) => {
    const id = prefix + label.toLowerCase();
    const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      setSelectedRadioValue(event.target.value);
    };

    const radioButtonProps: RadioButtonProps = {
      id,
      name: groupLabel,
      type: "radio",
      value: label,
      "aria-checked": label === selectedRadioValue,
      checked: label === selectedRadioValue,
      onChange,
    };

    const radioLabelProps: RadioLabelProps = {
      htmlFor: id,
    };

    return {
      radioButtonProps,
      radioLabelProps,
    };
  };

  return {
    selectedRadioValue,
    radioGroupWrapperProps,
    makeRadioButtonProps,
  } as UseRadioGroupReturn;
}
