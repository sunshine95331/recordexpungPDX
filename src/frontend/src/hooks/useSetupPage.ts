import { useState, useEffect } from "react";

export default function useSetupPage(pageName: string, scrollTop = false) {
  const [title, setTitle] = useState(pageName);
  const [shouldScrollTop, setShouldScrollTop] = useState(scrollTop);

  useEffect(() => {
    document.title = title + " - RecordSponge";
  }, [title]);

  useEffect(() => {
    window.scrollTo(0, 0);
  }, [shouldScrollTop]);

  return { title, setTitle, shouldScrollTop, setShouldScrollTop };
}
