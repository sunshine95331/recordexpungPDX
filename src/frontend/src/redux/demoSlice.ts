import { createSlice } from "@reduxjs/toolkit";
import { RootState } from "./store";

interface DemoState {
  isOn: boolean;
}

const initialState: DemoState = {
  isOn: false,
};

export const demoSlice = createSlice({
  name: "demo",
  initialState,
  reducers: {
    startDemo: (state) => {
      state.isOn = true;
    },
    stopDemo: (state) => {
      state.isOn = false;
    },
  },
});

export const { startDemo, stopDemo } = demoSlice.actions;

export const selectDemoStatus = (state: RootState) => state.demo.isOn;

export default demoSlice.reducer;
