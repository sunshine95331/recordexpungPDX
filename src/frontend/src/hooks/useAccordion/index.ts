// https://www.w3.org/WAI/ARIA/apg/patterns/accordion/

import React, { useState, useEffect, Dispatch } from "react";
import useDisclosure, {
  DisclosureButtonProps,
  DisclosureContentProps,
} from "../useDisclosure";

export type UseAccordionSection = ({
  id,
}: {
  id: string;
}) => {
  isExpanded: boolean;
  headerProps: DisclosureButtonProps;
  panelProps: DisclosureContentProps;
};
// Use useAccordion in a parent element and then pass
// useAccordionSection to each child element that will represent
// the accordion section that contains the heading and panel.
// Use buttons for headings to allow tabbing.
export default function useAccordion(idPrefix = "accordion") {
  const [openPanelId, setOpenPanelId] = useState("");
  const isExpandedSetters: {
    [id in string]: Dispatch<React.SetStateAction<boolean>>;
  } = {};

  // Do not use map iterators for id's. They can cause odd behavior.
  const useAccordionSection = ({ id }: { id: string }) => {
    const id_ = `${idPrefix}-panel-${id}`;
    const {
      disclosureIsExpanded: isExpanded,
      disclosureButtonProps: headerProps,
      disclosureContentProps: panelProps,
      setIsExpanded,
    } = useDisclosure({
      id: id_,
      callback: setOpenPanelId,
    });

    isExpandedSetters[id_] = setIsExpanded;

    return {
      isExpanded,
      headerProps,
      panelProps,
    };
  };

  useEffect(() => {
    Object.entries(isExpandedSetters).forEach(([id, setIsExpanded]) => {
      if (id !== openPanelId) {
        setIsExpanded(false);
      }
    });
    // eslint-disable-next-line
  }, [openPanelId]);

  return {
    useAccordionSection,
  };
}
