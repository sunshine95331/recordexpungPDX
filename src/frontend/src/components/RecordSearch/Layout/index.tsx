import React, { useState, useEffect } from "react";
import { useAppSelector } from "../../../redux/hooks";
import { selectStats } from "../../../redux/statsSlice";
import useRadioGroup from "../../../hooks/useRadioGroup";
import setUpScrollSpy from "../ExpandedView/setupScrollSpy";
import SearchPanel from "../SearchPanel";
import Status from "../Status";
import RecordSummary from "../Record/RecordSummary";
import Record from "../Record";
import Assumptions from "../Assumptions";
import { convertCaseNumberIntoLinks } from "../Record/util";
import ViewOptions from "../ExpandedView/ViewOptions";
import ExpandedView from "../ExpandedView";

function ErrorMessage({ message, idx }: { message: string; idx: number }) {
  const id = "record_error_" + idx;
  return (
    <p role="status" id={id} key={id} className="bg-washed-red mv3 pa3 br3 fw6">
      {convertCaseNumberIntoLinks(message)}
    </p>
  );
}

export const mainWrapper = "mw8 center f6 f5-l ";
export const panelClass = "bg-white shadow br3 ";
export const singlePanelClass = "pa4 mt3 ";

export default function Layout({
  topSection,
}: {
  topSection?: React.ReactElement;
}) {
  const record = useAppSelector((state) => state.search.record);
  const stats = useAppSelector(selectStats);
  const [showColor, setShowColor] = useState(true);
  const { selectedRadioValue, ...radioGroupProps } = useRadioGroup({
    label: "Summary overview sort options",
    initialValue: "Default",
  });
  const radioGroupOptions = ["Default", "Expanded"];
  const isExpandedView = selectedRadioValue === "Expanded";
  useEffect(setUpScrollSpy, [isExpandedView, record]);

  useEffect(() => {
    document.querySelector("html")?.classList.toggle("smooth-scroll");
    return () => {
      document.querySelector("html")?.classList.toggle("smooth-scroll");
    };
  });

  return (
    <main>
      <div className={mainWrapper + "ph2"}>
        {topSection}
        {!topSection && <h1 className="visually-hidden">Record Search</h1>}

        <section className={panelClass + singlePanelClass}>
          <SearchPanel />
        </section>

        <section className={panelClass + "mv3 ph3 pv1"}>
          <div className="flex flex-wrap items-center">
            <div className="f5 fw7 mr4">View Options</div>

            <ViewOptions
              optionLabels={radioGroupOptions}
              radioGroupProps={radioGroupProps}
              showColor={showColor}
              isExpandedView={isExpandedView}
              handleShowColorChange={() => setShowColor(!showColor)}
            />
          </div>
        </section>

        <Status />

        {record?.errors?.map((message: string, idx: number) => (
          <ErrorMessage key={idx} message={message} idx={idx} />
        ))}

        {!isExpandedView && (
          <>
            <section>
              <RecordSummary />
              <Record />
            </section>

            <Assumptions />
          </>
        )}
      </div>

      {isExpandedView && stats.totalCases > 0 && (
        <ExpandedView showColor={showColor} />
      )}
    </main>
  );
}
