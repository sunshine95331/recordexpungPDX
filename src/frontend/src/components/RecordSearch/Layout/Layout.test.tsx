import React from "react";
import { appRender } from "../../../test/testHelpers";
import { RECORD_LOADING } from "../../../redux/search/types";
import { useAppDispatch } from "../../../redux/hooks";
import Layout from ".";

beforeEach(() => {
  jest.useFakeTimers().setSystemTime(new Date(2023, 1, 6));
});

afterEach(() => {
  jest.useRealTimers();
});

it("can displays request errors", () => {
  const { asFragment } = appRender(<Layout />, "error");
  expect(asFragment()).toMatchSnapshot();
});

it("can display a loading spinner", () => {
  const RecordLoading = () => {
    useAppDispatch()({
      type: RECORD_LOADING,
      aliases: [],
      today: "",
    });
    return <Layout />;
  };

  const { asFragment } = appRender(<RecordLoading />);
  expect(asFragment()).toMatchSnapshot();
});
