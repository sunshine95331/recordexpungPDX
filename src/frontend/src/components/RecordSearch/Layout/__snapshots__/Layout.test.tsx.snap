// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can display a loading spinner 1`] = `
<DocumentFragment>
  <main>
    <div
      class="mw8 center f6 f5-l ph2"
    >
      <h1
        class="visually-hidden"
      >
        Record Search
      </h1>
      <section
        class="bg-white shadow br3 pa4 mt3"
      >
        <form
          class="mw7 center"
          novalidate=""
        >
          <div
            class="flex"
          >
            <div
              class="w-100 w-third-ns w-25-l mb3 pl2-r"
            >
              <label
                class="db mb1 fw6"
                for="today"
              >
                Expunge Date 
                <span
                  class="fw4 f6"
                >
                  mm/dd/yyyy
                </span>
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 "
                id="today"
                required=""
                type="text"
                value="2/6/2023"
              />
            </div>
          </div>
          <div
            class="flex flex-wrap items-end"
          >
            <div
              class="w-100 w-third-ns w-25-l mb3 "
            >
              <label
                class="db mb1 fw6"
                for="firstName"
              >
                First Name 
                <span
                  class="fw4 f6"
                />
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 br-0-ns br--left-ns relative"
                id="firstName"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="w-100 w-third-ns w-25-l mb3 "
            >
              <label
                class="db mb1 fw6"
                for="middleName"
              >
                Middle Name 
                <span
                  class="fw4 f6"
                >
                  (Optional)
                </span>
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 br0-ns"
                id="middleName"
                type="text"
                value=""
              />
            </div>
            <div
              class="w-100 w-third-ns w-25-l mb3 "
            >
              <label
                class="db mb1 fw6"
                for="lastName"
              >
                Last Name 
                <span
                  class="fw4 f6"
                />
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 bl-0-ns br--right-ns"
                id="lastName"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="w-100 w-third-ns w-25-l mb3 pl2-l"
            >
              <label
                class="db mb1 fw6"
                for="birthDate"
              >
                Date of Birth 
                <span
                  class="fw4 f6"
                >
                  mm/dd/yyyy
                </span>
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 "
                id="birthDate"
                type="text"
                value=""
              />
            </div>
            <div
              class="visually-hidden flex items-center pb1 mb3 ml3-ns ml0-l"
            >
              <span
                class=" visually-hidden fw5 bl bw2 b--blue bg-gray-blue-2 pa2 pr3 mr2 mb2"
              >
                1 Result
              </span>
              <button
                class="br2 bg-gray-blue-2 link hover-dark-blue mid-gray fw5 pa2 mb2"
                type="button"
              >
                <i
                  class="fas fa-times-circle pr1"
                />
                Remove
              </button>
            </div>
          </div>
          <div
            class="flex"
          >
            <button
              class="w4 tc br2 bg-gray-blue-2 link hover-dark-blue mid-gray fw5 pv3 ph3 mr2"
              type="button"
            >
              <i
                aria-hidden="true"
                class="fas fa-plus-circle pr1"
              />
              Alias
            </button>
            <button
              class="br2 bg-blue white bg-animate hover-bg-dark-blue db w-100 tc pv3 btn--search"
              type="submit"
            >
              <i
                aria-hidden="true"
                class="fas fa-search pr2"
              />
              <span
                class="fw7"
              >
                Search
              </span>
            </button>
          </div>
          <div
            class="black-70 w-100"
            role="alert"
          />
        </form>
      </section>
      <section
        class="bg-white shadow br3 mv2 ph3 pv1"
      >
        <div
          class="flex flex-wrap items-center"
        >
          <div
            class="f5 fw7 mr4"
          >
            View Options
          </div>
          <fieldset
            aria-label="Summary overview sort options"
            class="flex flex-wrap radio radio-sm"
            dir="ltr"
            role="radiogroup"
          >
            <p
              class="pr3"
            >
              <input
                aria-checked="true"
                checked=""
                id="summary-overview-sort-options-default"
                name="summary overview sort options"
                type="radio"
                value="Default"
              />
              <label
                for="summary-overview-sort-options-default"
              >
                Default
              </label>
            </p>
            <p
              class="pr3"
            >
              <input
                aria-checked="false"
                id="summary-overview-sort-options-expanded"
                name="summary overview sort options"
                type="radio"
                value="Expanded"
              />
              <label
                for="summary-overview-sort-options-expanded"
              >
                Expanded
              </label>
            </p>
          </fieldset>
        </div>
      </section>
      <section>
        <p
          class="bg-white shadow mv4 pa4 br3 fw6 tc"
        >
          <span
            class="spinner mr2"
          />
          Loading your search results...
        </p>
      </section>
      <section>
        <div
          class="tr"
        >
          <button
            class="inline-flex bg-white f6 fw5 br2 ba b--black-10 mid-gray link hover-blue pv1 ph2 mb3"
          >
            Enable Editing
          </button>
           
          <a
            class="link hover-blue "
            href="/manual#editing"
          >
            <i
              aria-hidden="true"
              class="gray hover-dark-blue fas fa-question-circle"
            />
            <span
              class="visually-hidden"
            >
              Editing Help
            </span>
          </a>
        </div>
      </section>
      <div
        class="bg-white shadow mb6 pa4 br3"
      >
        <h2
          class="fw6 mb3"
        >
          Assumptions
        </h2>
        <p
          class="mb3"
        >
          We are only able to access your public Oregon records.
        </p>
        <p
          class="mb2"
        >
          Your analysis may be different if you have had cases which were:
        </p>
        <ul
          class="lh-copy pl4 mw6 mb3"
        >
          <li
            class="mb2"
          >
            Previously expunged
          </li>
          <li
            class="mb2"
          >
            From States besides Oregon within the last ten years
          </li>
          <li
            class="mb2"
          >
            From Federal Court within the last ten years
          </li>
          <li
            class="mb2"
          >
            From local District Courts, e.g. Medford Municipal Court (not Jackson County Circuit Court) from within the last ten years
          </li>
        </ul>
        <p>
          <a
            class="link hover-blue underline"
            href="/manual#assumption1"
          >
            Learn more in the Manual
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`can displays request errors 1`] = `
<DocumentFragment>
  <main>
    <div
      class="mw8 center f6 f5-l ph2"
    >
      <h1
        class="visually-hidden"
      >
        Record Search
      </h1>
      <section
        class="bg-white shadow br3 pa4 mt3"
      >
        <form
          class="mw7 center"
          novalidate=""
        >
          <div
            class="flex"
          >
            <div
              class="w-100 w-third-ns w-25-l mb3 pl2-r"
            >
              <label
                class="db mb1 fw6"
                for="today"
              >
                Expunge Date 
                <span
                  class="fw4 f6"
                >
                  mm/dd/yyyy
                </span>
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 "
                id="today"
                required=""
                type="text"
                value="2/6/2023"
              />
            </div>
          </div>
          <div
            class="flex flex-wrap items-end"
          >
            <div
              class="w-100 w-third-ns w-25-l mb3 "
            >
              <label
                class="db mb1 fw6"
                for="firstName"
              >
                First Name 
                <span
                  class="fw4 f6"
                />
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 br-0-ns br--left-ns relative"
                id="firstName"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="w-100 w-third-ns w-25-l mb3 "
            >
              <label
                class="db mb1 fw6"
                for="middleName"
              >
                Middle Name 
                <span
                  class="fw4 f6"
                >
                  (Optional)
                </span>
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 br0-ns"
                id="middleName"
                type="text"
                value=""
              />
            </div>
            <div
              class="w-100 w-third-ns w-25-l mb3 "
            >
              <label
                class="db mb1 fw6"
                for="lastName"
              >
                Last Name 
                <span
                  class="fw4 f6"
                />
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 bl-0-ns br--right-ns"
                id="lastName"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="w-100 w-third-ns w-25-l mb3 pl2-l"
            >
              <label
                class="db mb1 fw6"
                for="birthDate"
              >
                Date of Birth 
                <span
                  class="fw4 f6"
                >
                  mm/dd/yyyy
                </span>
              </label>
              <input
                aria-invalid="false"
                class="w-100 br2 b--black-20 pa3 "
                id="birthDate"
                type="text"
                value=""
              />
            </div>
            <div
              class="visually-hidden flex items-center pb1 mb3 ml3-ns ml0-l"
            >
              <span
                class=" visually-hidden fw5 bl bw2 b--blue bg-gray-blue-2 pa2 pr3 mr2 mb2"
              >
                1 Result
              </span>
              <button
                class="br2 bg-gray-blue-2 link hover-dark-blue mid-gray fw5 pa2 mb2"
                type="button"
              >
                <i
                  class="fas fa-times-circle pr1"
                />
                Remove
              </button>
            </div>
          </div>
          <div
            class="flex"
          >
            <button
              class="w4 tc br2 bg-gray-blue-2 link hover-dark-blue mid-gray fw5 pv3 ph3 mr2"
              type="button"
            >
              <i
                aria-hidden="true"
                class="fas fa-plus-circle pr1"
              />
              Alias
            </button>
            <button
              class="br2 bg-blue white bg-animate hover-bg-dark-blue db w-100 tc pv3 btn--search"
              type="submit"
            >
              <i
                aria-hidden="true"
                class="fas fa-search pr2"
              />
              <span
                class="fw7"
              >
                Search
              </span>
            </button>
          </div>
          <div
            class="black-70 w-100"
            role="alert"
          />
        </form>
      </section>
      <section
        class="bg-white shadow br3 mv2 ph3 pv1"
      >
        <div
          class="flex flex-wrap items-center"
        >
          <div
            class="f5 fw7 mr4"
          >
            View Options
          </div>
          <fieldset
            aria-label="Summary overview sort options"
            class="flex flex-wrap radio radio-sm"
            dir="ltr"
            role="radiogroup"
          >
            <p
              class="pr3"
            >
              <input
                aria-checked="true"
                checked=""
                id="summary-overview-sort-options-default"
                name="summary overview sort options"
                type="radio"
                value="Default"
              />
              <label
                for="summary-overview-sort-options-default"
              >
                Default
              </label>
            </p>
            <p
              class="pr3"
            >
              <input
                aria-checked="false"
                id="summary-overview-sort-options-expanded"
                name="summary overview sort options"
                type="radio"
                value="Expanded"
              />
              <label
                for="summary-overview-sort-options-expanded"
              >
                Expanded
              </label>
            </p>
          </fieldset>
        </div>
      </section>
      <section>
        <p
          class="bg-light-gray mv4 pa4 br3 fw6"
        >
          No search results found.
        </p>
      </section>
      <p
        class="bg-washed-red mv3 pa3 br3 fw6"
        id="record_error_0"
        role="status"
      >
        first error
      </p>
      <p
        class="bg-washed-red mv3 pa3 br3 fw6"
        id="record_error_1"
        role="status"
      >
        second errorz
      </p>
      <section>
        <div
          class="bg-white shadow br3 mb3 ph3 pb3"
        >
          <div
            class="flex flex-wrap justify-end mb1"
          >
            <h2
              class="f5 fw7 mv3 mr-auto"
            >
              Search Summary
            </h2>
            <button
              class="ma2 nowrap mid-gray link fw6 br3 pv1 ph2 hover-blue"
            >
              <i
                aria-hidden="true"
                class="fas fa-bolt pr2"
              />
              Generate Paperwork
            </button>
            <button
              class="ma2 nowrap mid-gray link fw6 br3 pv1 ph2 hover-blue "
            >
              <i
                aria-hidden="true"
                class="fas fa-download pr2"
              />
              Summary PDF
            </button>
          </div>
          <div
            class="flex-ns flex-wrap"
          >
            <div
              class="w-100 w-two-thirds-l"
            >
              <h3
                class="bt b--light-gray pt2 mb3"
              >
                <span
                  class="fw7"
                >
                  Cases
                </span>
                 (0)
              </h3>
              <h3
                class="bt b--light-gray pt2 mb3"
              >
                <span
                  class="fw7"
                >
                  Charges
                </span>
                 (0)
                <p
                  class="f6 fw5 pt1"
                >
                  Excludes traffic violations
                </p>
              </h3>
              <div
                class="mb3"
              >
                <div
                  class="green bb b--light-gray lh-copy pb1"
                >
                  <span
                    class="fw7 mb2"
                  >
                    Eligible Now
                  </span>
                   
                </div>
                <ul
                  class="list"
                />
              </div>
            </div>
            <div
              class="w-100 w-33-l ph3-l mb3"
            >
              <div
                class="mb4"
              >
                <h3
                  class="fw7 bt b--light-gray pt2 mb3"
                >
                  Balance due by county 
                  <a
                    class=" gray link hover-blue underline"
                    href="/manual#paybalances"
                  >
                    <i
                      aria-hidden="true"
                      class="fas fa-question-circle link hover-dark-blue"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Learn more about balances
                    </span>
                  </a>
                </h3>
                <ul
                  class="mw5 list"
                >
                  None
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tr"
        >
          <button
            class="inline-flex bg-white f6 fw5 br2 ba b--black-10 mid-gray link hover-blue pv1 ph2 mb3"
          >
            Enable Editing
          </button>
           
          <a
            class="link hover-blue "
            href="/manual#editing"
          >
            <i
              aria-hidden="true"
              class="gray hover-dark-blue fas fa-question-circle"
            />
            <span
              class="visually-hidden"
            >
              Editing Help
            </span>
          </a>
        </div>
        <div
          class="mb3"
        >
          <ul
            class="list"
          />
        </div>
      </section>
      <div
        class="bg-white shadow mb6 pa4 br3"
      >
        <h2
          class="fw6 mb3"
        >
          Assumptions
        </h2>
        <p
          class="mb3"
        >
          We are only able to access your public Oregon records.
        </p>
        <p
          class="mb2"
        >
          Your analysis may be different if you have had cases which were:
        </p>
        <ul
          class="lh-copy pl4 mw6 mb3"
        >
          <li
            class="mb2"
          >
            Previously expunged
          </li>
          <li
            class="mb2"
          >
            From States besides Oregon within the last ten years
          </li>
          <li
            class="mb2"
          >
            From Federal Court within the last ten years
          </li>
          <li
            class="mb2"
          >
            From local District Courts, e.g. Medford Municipal Court (not Jackson County Circuit Court) from within the last ten years
          </li>
        </ul>
        <p>
          <a
            class="link hover-blue underline"
            href="/manual#assumption1"
          >
            Learn more in the Manual
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
